src/sagemaker_xgboost_container/metrics/custom_metrics.py

```diff
@@ -51,6 +53,24 @@ def accuracy(preds, dtrain):
 
 
 def f1(preds, dtrain):
+    """Compute f1 binary score. This can be used for binaryclassification training.
+
+    For more information see: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html
+
+    :param preds: Prediction values
+    :param dtrain: Training data with labels
+    :return: Metric name, f1 score
+    """
+    score = 0.0
+    if preds.size > 0:
+        labels = dtrain.get_label()
+        pred_labels = margin_to_class_label(preds)
+        pos_label = Counter(labels).most_common()[-1][0]  # using least frequent label as positive label
+        score = f1_score(labels, pred_labels, average='binary', pos_label=pos_label)
+    return 'f1', score
+
+
+def f1macro(preds, dtrain):
```

Comment:
Why not `f1_macro`? It follows the python code styling and also matches the metric name.

Reply:
Make totally sense, will change it in the next revision.